components/authz-service/storage/postgres/migration/sql/55_use_project_db_id_in_references_of_project_roles.up.sql

```diff
@@ -0,0 +1,102 @@
+BEGIN;
+-- break all references
+ALTER TABLE iam_role_projects
+    DROP CONSTRAINT iam_role_projects_project_id_fkey;
+
+ALTER TABLE iam_role_projects RENAME COLUMN project_id TO project_temp_id;
+ALTER TABLE iam_role_projects
+    ADD COLUMN project_id INTEGER REFERENCES iam_projects (db_id) ON DELETE CASCADE DEFERRABLE;
+UPDATE
+    iam_role_projects t
+SET
+    project_id = (
+        SELECT
+            db_id
+        FROM
+            iam_projects
+        WHERE
+            id = t.project_temp_id);
+ALTER TABLE iam_role_projects
+    DROP COLUMN project_temp_id;
+
+-- returns NULL if there's no associated projects; the call can decide
+-- (by coalesce'ing) if they want that to be {} or {(unassigned)}.
+CREATE OR REPLACE FUNCTION
+  role_projects(_role_id TEXT, OUT _project_ids TEXT[])
+  RETURNS TEXT[] AS $$
+
+  DECLARE
+  role_db_id INTEGER;
+  BEGIN
+      SELECT db_id INTO STRICT role_db_id
+      FROM iam_roles AS r
+      WHERE r.id=_role_id;
+      IF FOUND THEN
+          SELECT array_agg(p.id) INTO _project_ids
+          FROM iam_role_projects AS rp
+          LEFT JOIN iam_projects AS p
+          ON rp.project_id=p.db_id
+          WHERE rp.role_id=role_db_id;
```

Comment:
[nit] Some tabbing might make this a bit more readable.

Reply:
I've given all my trust to `pg_format` 😉